dynamic_programming/maximum_circular_subarray.cpp

```diff
@@ -0,0 +1,90 @@
+/**
+* @file
+* @brief C++ program for maximum contiguous circular sum problem using [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem)
+* @details
+* The idea is to modify Kadane’s algorithm to find a minimum contiguous subarray sum and the maximum contiguous subarray sum,
+* then check for the maximum value between the max_value and the value left after subtracting min_value from the total sum.
+* For more information, check [Geeks For Geeks](https://www.geeksforgeeks.org/maximum-contiguous-circular-sum/) explanation page.
+*/
+
+#include <cassert>     /// for assert
+#include <iostream>   /// for IO operations
+#include <vector>     /// for std::vector
+
+
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic Programming algorithms
+ */
+namespace dynamic_programming {
+/**
+ * @brief returns the maximum contiguous circular sum of an array
+ * 
+ * @param arr is the array/vector
+ * @return int which is the maximum sum
+ */
+int maxCircularSum(std::vector<int>& arr)
+{
+    // Edge Case
+    if (arr.size() == 1)
```

Comment:
We can capture the arr.size() in a variable since it is being used at multiple places. Can we handle arr of size 0 as well.

Reply:
The lookup time for the size is `O(1)` there is no real benefit to it unless you're talking about readability... but even then I think its more readable with `arr.size()` rather than `size` or `len`.